e2e/nomostest/testutils/hydrated_resources_validation.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"kpt.dev/configsync/e2e/nomostest"
+	"kpt.dev/configsync/e2e/nomostest/testpredicates"
+	"kpt.dev/configsync/pkg/metadata"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ValidateAllTenants validates all DRY resources under
+// '../testdata/hydration/kustomize-components' are reconciled.
+func ValidateAllTenants(nt *nomostest.NT, reconcilerScope, baseRelPath string, tenants ...string) {
+	nt.T.Logf("Validate resources are synced for all tenants %s", tenants)
+	ValidateNamespaces(nt, tenants, fmt.Sprintf("path: %s/namespace.yaml\n", baseRelPath))
+	for _, tenant := range tenants {
+		ValidateTenant(nt, reconcilerScope, tenant, baseRelPath)
+	}
+}
+
+// ValidateTenant validates the provided tenant resources under
+// '../testdata/hydration/kustomize-components' are reconciled.
```

Comment:
nit: These names seem a bit too generic given that the validation is specific to `kustomize-components`. Perhaps we could move these to a `kustomize_components` subpackage, such that the call would look like `kustomize_components.ValidateNamespaces` etc.

Reply:
I like it. The call looks self-description after moving into a specific subpackage. Thanks!